packages/main/cypress/specs/DateRangePicker.cy.tsx

```diff
@@ -0,0 +1,570 @@
+
+		cy.get("@dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realPress("F4");
+
+		cy.get("@calendarHeader")
+			.find("[data-ui5-cal-header-btn-next]")
+			.realClick();
+
+		cy.get("@calendarHeader")
+			.find("[data-ui5-cal-header-btn-month]")
+			.as("monthButton")
+			.should("have.text", "October");
+	});
+
+	it("startDateValue and endDateValue getters when only start date is present", () => {
+		cy.mount(<DateRangePicker value="27/09/2019" formatPattern="dd/MM/yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.then($datePicker => {
+				cy.wrap({
+					startDateValue: () => $datePicker[0].startDateValue,
+					endDateValue: () => $datePicker[0].endDateValue
+				})
+					.as("dates")
+					.invoke("startDateValue")
+					.then(date => {
+						const timestamp = date ? date.getTime() : new Date().getTime();
+						expect(new Date(timestamp)).to.deep.equal(new Date(2019, 8, 27));
+					});
+
+				cy.get("@dates")
+					.invoke("endDateValue")
+					.then(date => {
+						expect(date.toString()).to.be.equal("Invalid Date");
+					});
+			});
+	});
+
+	it("Picker popover should have accessible name", () => {
+		cy.mount(<DateRangePicker></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realPress("F4");
+
+		cy.get("@dateRangePicker")
+			.shadow()
+			.find("ui5-responsive-popover")
+			.should("have.attr", "accessible-name", "Choose Date Range");
+	});
+
+	it("Selected days: accessibility semantics", () => {
+		cy.wrap({ setLanguage })
+			.invoke("setLanguage", "en");
+
+		cy.mount(<DateRangePicker formatPattern="dd/MM/yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("09/06/2024 - 15/06/2024")
+			.realPress("Enter")
+			.realPress("F4");
+
+		cy.get<DateRangePicker>("@dateRangePicker")
+			.shadow()
+			.find("ui5-calendar")
+			.shadow()
+			.find("ui5-daypicker")
+			.shadow()
+			.find(".ui5-dp-root .ui5-dp-content div > .ui5-dp-item")
+			.then(days => {
+				const startSelectionDay = days[14];
+				const dayInBetween = days[15];
+				const endSelectionDay = days[20];
+
+				expect(startSelectionDay).to.have.attr("aria-selected", "true");
+				expect(dayInBetween).to.have.attr("aria-selected", "true");
+				expect(endSelectionDay).to.have.attr("aria-selected", "true");
+			});
+	});
+
+	it("Min and max dates are set without format-pattern by using ISO (YYYY-MM-dd) format", () => {
+		cy.wrap({ setLanguage })
+			.invoke("setLanguage", "bg");
+
+		cy.mount(<DateRangePicker minDate="2023-02-10" maxDate="2023-07-22"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.shadow()
+			.find("input")
+			.realClick()
```

Comment:
i think it will be good to ensure that the input is focused after the click and before we type